rpc/core/tx.go

```diff
@@ -19,36 +19,43 @@ import (
 // More: https://docs.tendermint.com/master/rpc/#/Info/tx
 func (env *Environment) Tx(ctx *rpctypes.Context, hash []byte, prove bool) (*ctypes.ResultTx, error) {
 	// if index is disabled, return error
-	if _, ok := env.TxIndexer.(*null.TxIndex); ok {
-		return nil, fmt.Errorf("transaction indexing is disabled")
-	}
 
-	r, err := env.TxIndexer.Get(hash)
-	if err != nil {
-		return nil, err
+	if isTxIndexingDisabled(env) {
+		return nil, errors.New("transaction indexing is disabled")
 	}
 
-	if r == nil {
-		return nil, fmt.Errorf("tx (%X) not found", hash)
-	}
+	for _, sink := range env.EventSinks {
+		if sink.Type() == indexer.KV {
+			r, err := sink.GetTxByHash(hash)
+			if err != nil {
+				return nil, err
+			}
 
-	height := r.Height
-	index := r.Index
+			if r == nil {
+				return nil, fmt.Errorf("tx (%X) not found", hash)
+			}
 
-	var proof types.TxProof
-	if prove {
-		block := env.BlockStore.LoadBlock(height)
-		proof = block.Data.Txs.Proof(int(index)) // XXX: overflow on 32-bit machines
+			height := r.Height
+			index := r.Index
+
+			var proof types.TxProof
+			if prove {
+				block := env.BlockStore.LoadBlock(height)
+				proof = block.Data.Txs.Proof(int(index)) // XXX: overflow on 32-bit machines
+			}
+
+			return &ctypes.ResultTx{
+				Hash:     hash,
+				Height:   height,
+				Index:    index,
+				TxResult: r.Result,
+				Tx:       r.Tx,
+				Proof:    proof,
+			}, nil
+		}
 	}
 
-	return &ctypes.ResultTx{
-		Hash:     hash,
-		Height:   height,
-		Index:    index,
-		TxResult: r.Result,
-		Tx:       r.Tx,
-		Proof:    proof,
-	}, nil
+	return nil, errors.New("could not find the event sink to support the tx query")
```

Comment:
```suggestion
	return nil, errors.New("transaction querying retrieval is not supported on this node. If this is incorrect, please enable it via config.toml")
```

Maybe something like this?

Reply:
I don't know how I feel about that error message. Tx indexing can be enabled, they just don't have KV enabled.

Reply:
I updated the error message. Is this more clear?
```
	return nil, errors.New("transaction querying is not supported on this node by the current settings." +
		"please check the tx-index section in the config.toml file if you don't expect to see this error")